dcm4che-core/src/main/java/org/dcm4che3/io/DicomOutputStream.java

```diff
@@ -170,6 +170,10 @@ public void writeHeader(int tag, VR vr, int len) throws IOException {
         ByteUtils.tagToBytes(tag, b, 0, bigEndian);
         int headerLen;
         if (!TagUtils.isItem(tag) && explicitVR) {
+            // Check to see if the size of the value to see if it is greater than 64Kib (8th bit is needed).
+            // If so, change the VR tag to unsigned before encoding the value.
```

Comment:
This comment explains what the code is doing (which should already be obvious from the code), but not the "why".
The issue description from Gunter explains it quite well:

> Encode Attributes whose known Value Representation is none of OB, OD, OF, OW, SQ, UC, UR or UT and whose value length exceeds 65534 (2^16-2) - and therefore cannot be encoded as a
> 16-bit unsigned integer in the Value Length Field defined for the known Value Representation - with Value Representation UN, according DICOM CP 1066 - ftp://medical.nema.org/medical/dicom/final/cp1066_ft.pdf

If you want to add a comment, that one might make more sense,

Reply:
To stay in sync with master, such changes (adding comments) should also be committed to master (via a PR)!